### DISCUSSION TITLE
Is serde working correctly in 0.5-rc1?

### DISCUSSION THREAD

**Discussion by ghost** [rwf2/Rocket#2026] [Questions] [answered]
I'm doing a small API and I have the following rocket version installed:

```toml
rocket = { version = "0.5.0-rc.1", features = ["secrets", "tls", "json"] }
```

However, when I'm importing the available methods and invoke the Serialization derive I get an error that reads: `error[E0463]: can't find crate for serde`

My following code is this:

```rust
use rocket::serde::{json::Json, Serialize};

#[derive(Serialize)]
```

Am I overlooking something here? :thinking:  Do I have to install `serde` and `serde_json` separately? If so, why? :scream:

**Comment by IniterWorker** [comment, marked as answer]
Hi @VentGrey,

It's more a Rust question.

> Rust's orphan rule requires that either the trait or the type for which you are implementing the trait must be defined in the same crate as the impl, so it is not possible to implement Serialize and Deserialize for a type in a different crate directly.

For more information, see : https://serde.rs/remote-derive.html


B. r

**Reply by ghost** [reply]
This was the answer. I did not have any idea about that :scream: 
Thank you so much for taking the time to respond to my question. :heart:

**Reply by Wulfheart** [reply]
@IniterWorker I am still not sure how to tackle this problem when using rocket in a single file.

Edit: This is valid:
```rust
#[derive(Deserialize)]
#[serde(crate = "rocket::serde")]
struct Test {
    test: String
}
```

**Reply by IniterWorker** [reply]
@Wulfheart is it a question or an affirmation?

**Reply by Wulfheart** [reply]
After the edit it is an affirmation.
